styx-scheduler-service/src/test/java/com/spotify/styx/StyxSchedulerTest.java

```diff
@@ -172,6 +190,79 @@ public void shouldFailToResetShardsOfResource() throws IOException {
     }
   }
 
+  @Test
+  public void shouldHandleWorkflowUnchanged() throws IOException {
+    when(storage.workflow(WORKFLOW_WITH_RESOURCES.id()))
+        .thenReturn(Optional.of(WORKFLOW_WITH_RESOURCES));
+    final Consumer<Workflow> consumer = StyxScheduler.workflowChanged(workflowInitializer, stats,
+        stateManager, storage, workflowConsumer);
+    consumer.accept(WORKFLOW_WITH_RESOURCES);
+    //noinspection unchecked
+    verify(stats).registerActiveStatesMetric(eq(WORKFLOW_WITH_RESOURCES.id()), any(Gauge.class));
+    verify(workflowInitializer).inspectChange(Optional.of(WORKFLOW_WITH_RESOURCES),
+        WORKFLOW_WITH_RESOURCES);
+    verify(workflowConsumer).accept(Optional.of(WORKFLOW_WITH_RESOURCES),
+        Optional.of(WORKFLOW_WITH_RESOURCES));
+  }
+
+  @Test
+  public void shouldHandleWorkflowChanged() throws IOException {
+    final Workflow workflow = Workflow.create(WORKFLOW_ID.componentId(),
+        HOURLY_WORKFLOW_CONFIGURATION_WITH_VALID_OFFSET);
+    when(storage.workflow(workflow.id())).thenReturn(Optional.of(workflow));
+    final Consumer<Workflow> consumer = StyxScheduler.workflowChanged(workflowInitializer, stats,
+        stateManager, storage, workflowConsumer);
+    consumer.accept(WORKFLOW_WITH_RESOURCES);
+    //noinspection unchecked
+    verify(stats).registerActiveStatesMetric(eq(WORKFLOW_WITH_RESOURCES.id()), any(Gauge.class));
```

Comment:
Use `any()` instead of `any(Gauge.class)` and remove the `//noinspection unchecked`